Vaccine Table updates

hoshmn opened the pull request:
Hey @Lane - this work addresses both of the Vaccine Table tickets
#351  
#241

Some notes/questions:

- the Total row stays at the top regardless of sorting, the BOP row comes next when sorted by Jurisdiction but sorts freely with other rows when sorted by a different column (see screenshot 2)
- #351 says to change copy "below" to "to the right". I changed to "here" since it can be either (see screenshot 1)
- ticket doesn't say what percentage columns should be called. If we want the table to fit comfortably in the 2 column layout at the md breakpoint, I don't think we can repeat "received at least one dose of vaccine" in all 4 column headers :) maybe we should experiment with two-tiered column headers (see screenshot 2), or a table title that allows for more succinct headers? 
- I'll fix the column widths once we decide on the above
- we had discussed making all table columns sortable for #241; I think the Facility column of the Facilities tables is the only column we could make sortable that isn't, but the column those tables are sorted by is tied to metric selection on the home page and group selection on the states pages, so we probably don't want to allow sorting by Facility
 
![Screen Shot 2021-05-26 at 11 14 47 AM](https://user-images.githubusercontent.com/18331999/119685910-a608cb80-bdfa-11eb-81d1-1e243458a088.png)

<img width="1211" alt="Screen Shot 2021-05-25 at 3 02 51 PM" src="https://user-images.githubusercontent.com/18331999/119682899-1a8e3b00-bdf8-11eb-8ffe-24c007d5f18d.png">

Lane reviewed (changes requested):
@hoshmn looking good! 🙌 some responses to your notes:

> the Total row stays at the top regardless of sorting, the BOP row comes next when sorted by Jurisdiction but sorts freely with other rows when sorted by a different column (see screenshot 2)

makes sense, i'm good with this.

> #351 says to change copy "below" to "to the right". I changed to "here" since it can be either (see screenshot 1)

thanks 👍

> ticket doesn't say what percentage columns should be called. If we want the table to fit comfortably in the 2 column layout at the md breakpoint, I don't think we can repeat "received at least one dose of vaccine" in all 4 column headers :) maybe we should experiment with two-tiered column headers (see screenshot 2), or a table title that allows for more succinct headers?

i like the two-tiered column headers, good call!  for the two-tiered column headings, can you take "received at least one dose of the vaccine" and do one of the following:
  - remove from the header, add a footnote to the table indicating the numbers represent people who have received at least one dose of the vaccine
  - break the text to a second line, set font weight to normal, use secondary text color

> we had discussed making all table columns sortable for #241; I think the Facility column of the Facilities tables is the only column we could make sortable that isn't, but the column those tables are sorted by is tied to metric selection on the home page and group selection on the states pages, so we probably don't want to allow sorting by Facility

right, that makes sense.  can we maintain some local state for the table sorted metric, so when they sort by facility:

- the table header updates do "Facilities by <ins>name</ins>▾"
- the table is sorted by facility name
- the global metric is not updated

and when sorting all other metrics that are available, update the global metric.

hoshmn commented:
Thanks for the feedback @Lane ! Added some updates

- added (93161cd) a small fix for the header/content alignment part of #309. I commented about the vaccine table total part on Slack; seems like this should be updated to reflect the added rates
- added (9eea062) the vaccine top level headers and set some widths on the columns
- added (37838f5) the ability to sort the facilities tables by name (both home and state pages). I thought I'd check back in before altering the metric selection dropdown above the table. 
     - **State page facilities table**: the only way you know which metric is displaying on the map is by that metric selection dropdown (unless you look in the tiny legend in the corner -- see screenshot). It also feels confusing to add an option to the metric selection ("name") that doesn't actually change the metric displaying on the map. Maybe the underlined table header is enough to indicate the facilities are sorted by name, and the metric selection should stay showing the active metric? (Worth noting that, even if we do this, the user will still only know which _group_ is selected by looking at the tiny legend, since it used to be indicated by the underlining of residents/staff).
     - **Home page facilities table**: doesn't have the same issues with user not knowing the group (always residents), and the active metric is still shown above the map. Still maybe feels confusing to add a non-metric option to the dropdown, though? Could make it something like "Facilities with the highest <ins>cumulative cases▾</ins> by name" ? 

<img width="1103" alt="Screen Shot 2021-05-27 at 12 03 11 PM" src="https://user-images.githubusercontent.com/18331999/119866613-d5d6d280-bed1-11eb-9cf5-ac0666ee4645.png">

hoshmn commented on packages/gatsby-ucla-site/content/en.json:
```diff
@@ -16,12 +16,13 @@
   "total": "Total",
   "vaccine_subtitle": "This table reflects data publicly reported by correctional agencies about the administration of COVID-19 vaccines to incarcerated people and staff in their facilities. Agencies differ in the number and definitions of variables they report about vaccination, but, to the best of our knowledge, the numbers reported here refer to the number of people who have received <u>at lea ...
-  "vaccine_note": "Vaccination rates refer to the number of people who have received at least one dose as a percentage of population. The population for incarcerated people is from January 2021, and the population for staff is the most recent data available. Many of the staff population denominators come from the Marshall Project’s data.",
+  "vaccine_note": "Vaccination totals refer to the number of people who have received at least one dose of vaccine. The population value used to calculate the vaccination rate for incarcerated people is from January 2021, and the population value used for staff is the most recent data available. Many of the staff population denominators come from the Marshall Project’s data.",
```
the "one dose" is explained in the vaccine_subtitle, so maybe it's redundant to include here?

hoshmn commented on packages/gatsby-ucla-site/content/en.json:
```diff
@@ -16,12 +16,13 @@
-  "initiated_percent": "As a percentage of the population",
+  "residents_initiated": "Incarcerated people",
+  "staff_initiated": "Staff",
+  "initiated_percent": "Rate",
```
changed to Rate rather than Percentage to bring into line with other metrics

Lane commented:
**State page facilities table**

ah, good point. for now, let's go with what you're suggesting here: 

> underlined table header is enough to indicate the facilities are sorted by name, and the metric selection should stay showing the active metric

ultimately, i think we need to reconsider the design of this section to better label what the map is showing and maybe separate controls from the header.

**Home page facilities table**

> Still maybe feels confusing to add a non-metric option to the dropdown, though? Could make it something like "Facilities with the highest cumulative cases▾ by name" ?

can we keep the "name" option out of the dropdown?  only update the title to include name if they have specifically sorted the table by name.  i think we want to avoid a title like "Facilities with the highest cumulative cases▾ by name" because the table is not showing the facilities with the highest cumulative cases when it is sorted by facility name.

hoshmn commented:
@Lane Cool, so I left state pages as-is (they were already set up according to my suggestion). Added the rest of the logic for the home page facilities table. All in all it's a significant amount of added code/complexity to make sortable by facility name, but I guess it is a helpful feature that some users might expect.
